### PR TITLE
FEA-2496: Fixed Cascade Indexing

### LINKED ISSUE #80: Cascade values index incorrect reference
Given the following dart setup:
```dart
class Foo() {
  int bar;
}

void main() {
  final value = 1;
  Foo()..bar = value
}
```

running index/snapshot, results in:

```dart
    Foo()..near = value;
//  ^^^ reference scip-dart pub dart_test 1.0.0 lib/`other.dart`/Foo#
//                ^^^^^ reference scip-dart pub dart_test 1.0.0 lib/`other.dart`/Foo#near.
```

Cascade on primitives also ignores indexing on the value

```dart
    Foo()..value = false;
//  ^^^ reference scip-dart pub dart_test 1.0.0 lib/`other.dart`/Foo#
```

### PR DESCRIPTION
**Pull request by matthewnitschke-wk**
# [FEA-2496](https://jira.atl.workiva.net/browse/FEA-2496)
Issue Status

Closes #80

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/scip_visitor.dart [hunk 1]
```diff
@@ -115,9 +115,12 @@ class ScipVisitor extends GeneralizingAstVisitor {
...
-    if (node.parent is CompoundAssignmentExpression) {
-      final assignmentNode = node.parent as CompoundAssignmentExpression;
-      element = assignmentNode.readElement ?? assignmentNode.writeElement;
+    if (element == null) {
+      final assignmentExpr =
+          node.thisOrAncestorOfType<CompoundAssignmentExpression>();
+      if (assignmentExpr == null) return;
+
+      element = assignmentExpr.readElement ?? assignmentExpr.writeElement;
     }
```
Comment (matthewnitschke-wk) [inline lib/src/scip_visitor.dart:124]: Given the following dart code:
```dart
Foo()..value = 1;
```

This is what the generated ast results in

```
Foo()..value = 1; (ExpressionStatement)
        Foo()..value = 1 (CascadeExpression)
                Foo() (MethodInvocation)
                        Foo (SimpleIdentifier)
                        () (ArgumentList)
                ..value = 1 (AssignmentExpression) <----- (extends from CompoundAssignmentExpression)
                        ..value (PropertyAccess)
                                value (SimpleIdentifier)
                        1 (IntegerLiteral)
```

Given the above code, `value (SimpleIdentifier)` is the case that makes it to this conditional. If we refer to the ast, the parent of this element is `PropertyAccess`, not `CompoundAssignmentExpression`, which tells us this expression can occur at multiple depths, not just the parent

We can solve this by just searching for the closest node of the specific type, and treat that as the assignment expression we care about

**Review by kirkrichardson-wk — APPROVED** [review]

**Comment by matthewnitschke-wk** [comment]
QA +1

- [x] Verified that snapshots resulted in correct results

**Comment by matthewnitschke-wk** [comment]
🚀 @Workiva/release-management-p 🚢

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**